tests/test_spec.py

```diff
@@ -98,84 +98,84 @@ def test_resources(self):
-             Blob(self.spec, "source1", "ocaml-cohttp-init",
+            [Blob(self.spec, "http://elsewhere", "link1"),
+             Blob(self.spec, "ocaml-cohttp-init",
                   "tests/data/ocaml-cohttp.spec"),
-             Blob(self.spec, "source2", "ocaml-cohttp-service",
+             Blob(self.spec, "ocaml-cohttp-service",
                   "tests/data/ocaml-cohttp.spec"),
-             Blob(self.spec, "source3", "http://additional", "link2"),
-             Blob(self.spec, "patch0", "cohttp0.patch",
+             Blob(self.spec, "http://additional", "link2"),
+             Blob(self.spec, "cohttp0.patch",
                   "tests/data/ocaml-cohttp.spec"),
-             Blob(self.spec, "patch1", "http://a.n.other", "link3"),
-             Blob(self.spec, "patch3", "http://extra", "link4")])
+             Blob(self.spec, "http://a.n.other", "link3"),
+             Blob(self.spec, "http://extra", "link4")])
 
     def test_resources_extras(self):
         """Package source paths and URLs are correct"""
-        self.spec.add_archive("archive0", "http://foo/patches.tar",
+        self.spec.add_archive(0, "http://foo/patches.tar",
                               "link1", "SOURCES/")
-        self.spec.add_patchqueue("patchqueue0", "http://foo/patchqueue.tar",
+        self.spec.add_patchqueue(0, "http://foo/patchqueue.tar",
                                  "link1", "PATCHES/")
         self.assertEqual(
             self.spec.resources(),
-            [Blob(self.spec, "source0",
+            [Blob(self.spec,
                   "https://github.com/mirage/ocaml-cohttp/archive/"
                   "ocaml-cohttp-0.9.8/ocaml-cohttp-0.9.8.tar.gz",
                   "tests/data/ocaml-cohttp.spec"),
-             Blob(self.spec, "source1", "ocaml-cohttp-init",
+             Blob(self.spec, "ocaml-cohttp-init",
                   "tests/data/ocaml-cohttp.spec"),
-             Blob(self.spec, "source2", "ocaml-cohttp-service",
+             Blob(self.spec, "ocaml-cohttp-service",
                   "tests/data/ocaml-cohttp.spec"),
-             Blob(self.spec, "patch0", "cohttp0.patch",
+             Blob(self.spec, "cohttp0.patch",
                   "tests/data/ocaml-cohttp.spec"),
-             Blob(self.spec, "patch1", "cohttp1.patch",
+             Blob(self.spec, "cohttp1.patch",
                   "tests/data/ocaml-cohttp.spec"),
-             Archive(self.spec, "archive0", "http://foo/patches.tar", "link1",
+             Archive(self.spec, "http://foo/patches.tar", "link1",
                      "SOURCES/"),
-             Patchqueue(self.spec, "patchqueue0", "http://foo/patchqueue.tar",
+             Patchqueue(self.spec, "http://foo/patchqueue.tar",
                         "link1", "PATCHES/")])
 
     def test_resource(self):
         """URLs for individual resources are correct"""
         self.assertEqual(
             self.spec.resource("path/to/ocaml-cohttp-0.9.8.tar.gz"),
-            Blob(self.spec, "source0",
```

Comment:
Could you please make one of these tests add 'source' with no digit, to test the special case in `_parse_name`